core/src/main/scala/org/elasticmq/actor/queue/InternalMessage.scala

```diff
@@ -83,15 +83,16 @@ case class InternalMessage(
 }
 
 object InternalMessage {
+  def from(newMessageData: NewMessageData, queueData: QueueData, sequenceNumber: BigInt): InternalMessage = {
+    val attributes = updateSequenceNumberAttribute(newMessageData, queueData, sequenceNumber)
```

Comment:
doesn't this assign a sequence number to every message, instead of only FIFO messages?

Reply:
No it doesn't. It creates number for every message in queue but doesn't assign it. If user sent `SequenceNumber` in attribute it will delete it.

Reply:
moving method fixed this